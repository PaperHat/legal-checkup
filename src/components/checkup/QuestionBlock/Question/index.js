import { connect } from 'react-redux'
import React from 'react'
import { PoseGroup } from 'react-pose'

import { getActiveQuestion } from '../../../../state/selectors'

import StyledQuestion from './style'

function mapStateToProps ({ checkup }) {
  const { body: children, id: key } = getActiveQuestion(checkup)
  console.log(key)

  return { children, key }
}

const Question = ({ children, key, format }) => {
  return (
    <PoseGroup>
      {children && (
        <StyledQuestion key={'h'} format={format}>
          {children}
        </StyledQuestion>
      )}
    </PoseGroup>
  )
}

export default connect(mapStateToProps)(Question)
