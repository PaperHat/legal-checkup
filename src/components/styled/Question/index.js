import styled from 'styled-components'

const Question = styled.li`
  margin: 0.2rem 0.1rem;
  padding: 0.3rem 0.5rem;
  border-radius: 0.2rem;

  ${({ active, answered, disabled }) => {
    let styles = ''

    if (active) {
      styles += `
        color: whitesmoke;
        background-color: gray;
      `
    }

    if (active || answered) {
      styles += `
        cursor: pointer;
        &:hover {
          background-color: gray;
          color: black;
        }
      `
    }

    if (disabled) {
      styles += `
        color: gray;
      `
    }

    return styles
  }};
`

Question.displayName = 'StyledQuestion'

export default Question
