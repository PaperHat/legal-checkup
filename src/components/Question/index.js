import React from 'react'

import { StyledQuestion } from '../styled'

export default function Question ({ answered, current, question }) {
  if (question === current) {
    return (
      <StyledQuestion key={question} active>
        {question}
      </StyledQuestion>
    )
  }

  if (question <= answered) {
    return (
      <StyledQuestion key={question} answered>
        {question}
      </StyledQuestion>
    )
  }

  return (
    <StyledQuestion key={question} disabled>
      {question}
    </StyledQuestion>
  )
}
