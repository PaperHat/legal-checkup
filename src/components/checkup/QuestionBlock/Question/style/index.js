import styled from 'styled-components'
import posed from 'react-pose'

import { DESKTOP, MOBILE, TABLET } from '../../../../../constants'
import styledMap from 'styled-map'

const Question = styled.div`
  font-family: 'Museo Sans Cyrl 500', sans-serif;
  font-weight: 500;
  text-align: center;
  margin: 0px auto;
  ${styledMap('format', {
    [DESKTOP]: `
  font-size: 32px;
  line-height: normal;
  padding-top: 58px;`,
    [MOBILE]: `
  font-size: 25px;
  line-height: 1.2;
  padding-top: 12px;`,
    [TABLET]: `
  font-size: 32px;
  line-height: normal;
  padding-top: 58px;`
  })}
`
Question.displayName = 'StyledQuestion'

const PosedQuestion = posed(Question)({
  enter: {
    opacity: 1
  },
  exit: {
    opacity: 0
  }
})

export default PosedQuestion
