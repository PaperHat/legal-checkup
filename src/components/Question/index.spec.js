import React from 'react'
import { shallow } from 'enzyme'

import Question from '.'

describe('components:Question', () => {
  it('renders the Question and CSS properties properly when active', () => {
    expect(
      toJson(
        shallow(
          <Question answered={2} current={2} question={2} total={3}>
            Question
          </Question>
        )
      )
    ).toMatchSnapshot()
  })

  it('renders the Question and CSS properties properly when answered', () => {
    expect(
      toJson(
        shallow(
          <Question answered={2} current={2} question={1} total={3}>
            Question
          </Question>
        )
      )
    ).toMatchSnapshot()
  })

  it('renders the Question and CSS properties properly when not current or answered', () => {
    expect(
      toJson(
        shallow(
          <Question answered={2} current={2} question={3} total={3}>
            Question
          </Question>
        )
      )
    ).toMatchSnapshot()
  })
})
