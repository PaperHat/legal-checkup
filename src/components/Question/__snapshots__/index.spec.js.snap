// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components:Question renders the Question and CSS properties properly when active 1`] = `
<StyledQuestion
  active={true}
  key="2"
>
  2
</StyledQuestion>
`;

exports[`components:Question renders the Question and CSS properties properly when answered 1`] = `
<StyledQuestion
  answered={true}
  key="1"
>
  1
</StyledQuestion>
`;

exports[`components:Question renders the Question and CSS properties properly when not current or answered 1`] = `
<StyledQuestion
  disabled={true}
  key="3"
>
  3
</StyledQuestion>
`;
